src/Capture/X11.hs

```diff
@@ -120,3 +126,36 @@ myFetchName d w = do
         bracket getProp (xFree . tp_value) extract
             `catchIOError` \_ -> return ""
 
+-- | Check active screen saver using the X11 Screen Saver extension.
+--
+-- This most likely only works with the simple built-in screen saver
+-- configured using @xset s@. Screen savers/lockers such as xscreensaver,
+-- xsecurelock, i3lock, etc. work differently.
```

Comment:
Ah, too bad. I was hoping this simple interface sufficies, and we’d not have to worry about dbus dependencies,  error handling, and the cost of doing dbus calls on each sample :-/

Reply:
Yeah, it's quite disappointing. Doing 5 dbus calls on each sample to discard 4 of them is bad. Caching which one works and which one just returns errors would be possible, but the code will get considerably more complex. :-(

Reply:
Maybe submit patches to the screensavers to set the X11 screen saver satus correctly? :-)